packages/x-components/src/x-modules/search/store/emitters.ts

```diff
@@ -23,6 +23,25 @@ export const searchEmitters = createStoreEmitters(searchXStoreModule, {
       return !!newValue && !!oldValue;
     }
   },
+  SearchResponseProvided: {
+    selector: (state, getters) => {
+      return {
+        request: getters.request!,
+        status: state.status,
+        banners: state.banners,
+        facets: state.facets,
+        partialResults: state.partialResults,
+        promoteds: state.promoteds,
+        queryTagging: state.queryTagging,
+        redirections: state.redirections,
+        results: state.results,
+        spellcheck: state.spellcheckedQuery,
+        totalResults: state.totalResults
+      };
+    },
+    filter: (newValue, oldValue) =>
+      oldValue.status === 'loading' && newValue.status !== oldValue.status
```

Comment:
As the selector includes the request object and that object changes before the request itself is performed (`UserAcceptedAQuery`), the best way to emit the event only when the response is received is by using the`status`.